src/gui/qgsmapcanvas.cpp

```diff
@@ -1038,6 +1031,36 @@ void QgsMapCanvas::showContextMenu( QgsMapMouseEvent *event )
   menu.exec( event->globalPos() );
 }
 
+void QgsMapCanvas::notifyRendererErrors( const QgsMapRendererJob::Errors &errors )
+{
+  const QDateTime currentTime = QDateTime::currentDateTime();
+
+  // remove errors too old
+  for ( const QgsMapRendererJob::Error &error : errors )
+  {
+    const QString errorKey = error.layerID + ':' + error.message;
+    if ( mRendererErrors.contains( errorKey ) )
+    {
+      const QDateTime sameErrorTime = mRendererErrors.value( errorKey );
+
+      if ( sameErrorTime.secsTo( currentTime ) < 60 )
+        continue;
+    }
+
+    mRendererErrors[errorKey] = currentTime;
+
+    QString message = error.message;
+    const QRegularExpression regEx( QStringLiteral( "(https?:\\/\\/+[\\/\\{\\}\\?=a-zA-Z0-9_.~-]*)" ), QRegularExpression::CaseInsensitiveOption );
+    QRegularExpressionMatch match = regEx.match( message );
+    if ( match.hasMatch() )
+      message.replace( regEx, "<a href=\"\\1\">\\1</a>" );
+
+    QgsMapLayer *layer = QgsProject::instance()->mapLayer( error.layerID );
+    emit renderErrorOccurred( message, layer );
+    QgsMessageLog::logMessage( error.layerID + " :: " + message, tr( "Rendering" ) );
```

Comment:
```suggestion
    if ( QgsMapLayer *layer = QgsProject::instance()->mapLayer( error.layerID ) )
    {
      emit renderErrorOccurred( message, layer );
      QgsMessageLog::logMessage( error.layerID + " :: " + message, tr( "Rendering" ) );
    }
```

Just in case!

Also I think here the QgsMessageLog call can be removed, as all warnings pushed to the messagebar are automatically logged to QgsMessageLog. Are you seeing duplicate log messages?